### PR TITLE
IAM-573-Create Demo Charm for Admin UI

### PR DESCRIPTION
**Pull request by bencekov**
This pr includes the demo charm for Admin UI. Relations with the other Identity platform components will come after this pr is merged. At this stage, the charm can be integrated with other components via juju configs.

### CODE REVIEW

**Review by nsklikas — COMMENTED** [review]
Can you please switch the comments from `""""""` to `#`? Let's use the first for doc-strings and the latter only for in-line comments.

File: .github/.jira_sync_config.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# From https://github.com/canonical/gh-jira-sync-bot#client-side-configuration
+settings:
+  jira_project_key: "IAM"
+  status_mapping:
+    opened: Untriaged
+    closed: done
+  components:
+    - Login UI
```
[the remark this answers is not in the document]
Comment (bencekov) [reply .github/.jira_sync_config.yaml]: done

File: src/charm.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        if self._tracing_ready:
+            container["environment"]["OTEL_HTTP_ENDPOINT"] = self._tracing_endpoint_info_http
+            container["environment"]["OTEL_GRPC_ENDPOINT"] = self._tracing_endpoint_info_grpc
+            container["environment"]["TRACING_ENABLED"] = True
+
+        # Define Pebble layer configuration
+        pebble_layer = {
+            "summary": "Pebble Layer for Identity Platform Admin UI",
+            "description": "Pebble Layer for Identity Platform Admin UI",
+            "services": {self._container_name: container},
```
[the remark this answers is not in the document]
Comment (bencekov) [reply src/charm.py]: done

File: src/charm.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+    """Charm event handlers"""
+
+    def _on_admin_ui_pebble_ready(self, event: WorkloadEvent) -> None:
+        """Define and start a workload using the Pebble API."""
+        if not self._container.can_connect():
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting to connect to admin-ui container")
+            return
+
+        """Makes sure the directory for the logfile exists"""
```
[the remark this answers is not in the document]
Comment (bencekov) [reply src/charm.py]: good catch. done.

File: src/charm.py [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+class IdentityPlatformAdminUiOperatorCharm(CharmBase):
+    """Identity Platform Admin Ui charm class."""
+
+    def __init__(self, *args):
+        """Initialize Charm."""
+        super().__init__(*args)
+
+        """Charm properties"""
+        self._container_name = "admin-ui"
```
[the remark this answers is not in the document]
Comment (bencekov) [reply src/charm.py]: done

**Review by nsklikas — APPROVED** [review]

**Pull request merged**